examples/arithmetics/src/interpreter.ts

```diff
@@ -0,0 +1,88 @@
+import { AbstractDefinition, Definition, Evaluation, Expression, isAddition, isDefinition, isDivision, isEvaluation, isFunctionCall, isMultiplication, isNumberLiteral, isSubtraction, Module, Statement } from './language-server/generated/ast';
+
+export class ArithmeticsInterpreter {
+    // variable name --> value
+    private context: Map<string, number | Definition>;
+    // expression --> value
+    private res = new Map<string, number>();
+
+    constructor(context: Map<string, number | Definition> = new Map<string, number | Definition>()) {
+        this.context = context;
+    }
+
+    public eval(module: Module): Map<string, number> {
+        module.statements.forEach(stmt => this.evalStatement(stmt));
+        return this.res;
+    }
+
+    private evalStatement(stmt: Statement): void {
+        if (isDefinition(stmt)) {
+            this.evalDefinition(stmt);
+        } else if (isEvaluation(stmt)) {
+            this.evalEvaluation(stmt);
+        }
+    }
+
+    private evalDefinition(def: Definition): void {
+        this.context.set(def.name, def.args.length > 0 ? def : this.evalExpression(def.expr));
+    }
+
+    private evalEvaluation(evaluation: Evaluation): void {
+        const expr = evaluation.expression;
+        if (expr.$cstNode?.text) {
+            this.res.set(expr.$cstNode?.text, this.evalExpression(expr));
```

Comment:
I'd propose to use the `evaluation` as map key and leave the printing of the expression text to the caller (the CLI `eval` command handler).